### ISSUE TITLE
💅 False negative `noUnusedVariables` in Astro file

### ISSUE THREAD

**Issue by 1v3n** [biomejs/biome#2863]
### Environment information

Hey,

inside an Astro file I get an false noUnusedVariables error. 

My biome setup for that:

```json

"rules": {
  "recommended": true,
  "correctness": {
    "noUnusedVariables": "error"
  }
}
```

index.astro
```astro
---
const { title } = Astro.props;
---

<span>{title}</span>

```

Causes 
```shell
 ✖ This variable is unused.
  
  > 2 │ const { title } = Astro.props;
      │         ^^^^^
    3 │ 
  
  ℹ Unused variables usually are result of incomplete refactoring, typos and other source of bugs.
  
```

### Rule name

noUnusedVariables

### Playground link

https://codesandbox.io/p/devbox/biome-bug-lkjq3m

### Expected result

I would not expect an error, as the variable is used In the JSX part of the Astro file.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by 1v3n** [comment]
Biome doesn't fully support Astro files, therefore I needed to set this:

```json
 "globals": ["Astro"]
```
[--> Source](https://biomejs.dev/internals/language-support/)

**Comment by motss** [comment]
@1v3n what do you set this?

```ts
 "globals": ["Astro"]
```

**Comment by ematipico** [comment]
> @1v3n what do you set this?
> 
> ```ts
>  "globals": ["Astro"]
> ```

`javascript.globals`: https://biomejs.dev/reference/configuration/#javascriptglobals

**Comment by castarco** [comment]
This problem is not related with `Astro` not being found by Biomejs. It's something entirely different.

**Comment by juxuanu** [comment]
I am still getting this issue with the default template of a new Astro project and the latest Biome.

```astro
---
import Welcome from "../components/Welcome.astro";
import Layout from "../layouts/Layout.astro";

// Welcome to Astro! Wondering what to do next? Check out the Astro documentation at https://docs.astro.build
// Don't want to use any of this? Delete everything in this file, the `assets`, `components`, and `layouts` directories, and start fresh.
---

<Layout>
	<Welcome />
</Layout>
```
 complains about `Welcome` and `Layout` being unused (triggers rule `lint/correctness/noUnusedImports`)
